app/api/dao/user.py

```diff
@@ -357,11 +355,11 @@ def get_achievements(user_id):
         """
         user = UserModel.find_by_id(user_id)
         all_relations = user.mentor_relations + user.mentee_relations
-        tasks = []
-        for relation in all_relations:
-            tasks += relation.tasks_list.tasks
-        achievements = [task for task in tasks if task.get("is_done")]
-        return achievements
+        # [relation.task_list...] returns an array with an array of array,
+        # that's why we need to take the first element
+        tasks = [relation.tasks_list.tasks for relation in all_relations][0]
+        achievements = filter(lambda task: task.get("is_done"), tasks)
+        return list(achievements)
```

Comment:
I suggest we let the previous code be because it is already using list comprehension and is quite readable that way. The aim should be optimised while maintaining readability  :)
IMHO this change here might be an overkill.

Reply:
@sara-02, you're absolutely right. I thought about this, but then I also thought that the task wanted to replace all of the occurrences, but I've removed this change :)